src/pages/ChangePassword.ts

```diff
@@ -0,0 +1,154 @@
+import ProfileForm from '../components/form/profile-form';
+import Input from '../components/input';
+import List from '../components/list';
+import LineInput from '../components/lineinput';
+import Button from '../components/button';
+import {REG_EXP, ERROR_MSG, Field, FieldBlock} from '../utils/validationConst';
+import LeftNav from '../components/leftnav';
+import Avatar from '../components/avatar';
+import { PATH } from '../router/paths';
+import userApi from '../api/user-api';
+import Router from '../router';
+import { User } from '../core/types';
+import Form from '../components/form';
+
+export default class ChangePassword extends List {
+	router = new Router('#root');
+	constructor(props: {user?: User} = {}) {
+
+		const classAvatar = 'avatar';
+		const avatarProps = {};
+		const avatar: Avatar = new Avatar(
+			avatarProps,
+			'div',
+			classAvatar
+		);
+		
+		const fields: Field[] = [
+				{
+					type: 'password', 
+					label: 'Старый\u00A0пароль', 
+					name: 'oldPassword', 
+					value: '', 
+					valid: REG_EXP.PASSWORD, 
+					errorMsg: ERROR_MSG.PASSWORD, 
+					autocomplete: 'current-password'
+				},
+				{
+					type: 'password', 
+					label: 'Новый\u00A0пароль', 
+					name: 'newPassword', 
+					value: '', 
+					valid: REG_EXP.PASSWORD, 
+					errorMsg: ERROR_MSG.PASSWORD
+				},
+				{
+					type: 'password', 
+					label: 'Повторите\u00A0новый\u00A0пароль', 
+					name: 'confirmPassword', 
+					value: '', 
+					valid: REG_EXP.PASSWORD, 
+					errorMsg: ERROR_MSG.CONFIRM_NEW_PASSWORD
+				},
+				{
+					type: 'hidden', 
+					name: 'login', 
+					value: props?.user?.login, 
+					autocomplete: 'login'
+				},
+			];
+		
+		const inputArr: Input[] = fields.map(({type, name, value, autocomplete}, i: number)=>{
+			return new Input(
+				{attr: {type, id: name, name, value, autocomplete, autofocus: (i===0)}},
+				'input', 
+				'form__input form__input__right'
+				);
+		});
+		
+		const lineInputArr: Input[]|LineInput[] = fields.map(({type, label, name, valid, errorMsg}, i)=>{
+			const lineinput: FieldBlock = {label, input: inputArr[i]};
+	
+			const fieldvalid = (name==='confirm_password') ? ()=>(
```

Comment:
Сразу почему-то не обратил внимание на именование переменных. Хорошим тоном считается использование одного стиля именования, например pascalCase
```suggestion
			const lineInput: FieldBlock = {label, input: inputArr[i]};
	
			const fieldValid = (name==='confirm_password') ? ()=>(
```

Reply:
Исправил